### DISCUSSION TITLE
Parse `Bytes` from `RawForm`

### DISCUSSION THREAD

**Discussion by maxwell8888** [tokio-rs/axum#1948] [Q&A] [answered]
When using `RawForm` like so from the docs:
```rust
async fn handler(RawForm(form): RawForm) {}
let app = Router::new().route("/", get(handler));
```
`form` is a `Bytes`. I want to parse the `Bytes` into a usable form like a vector of tuples or HashMap. Before I add another library to the project for this, I was wondering is there is any functionality within Axum to parse the `Bytes`?

**Comment by davidpdrsn** [comment, marked as answer]
You either need to use `Form` or handle it on your own (perhaps via another library)

**Comment by maxwell8888** [comment]
For anybody else with the same problem I ended up using `form_urlencoded` which does the job.
